tanner/api.py

```diff
@@ -38,3 +100,52 @@ def __init__(self):
             for (i, val) in enumerate(query_res):
                 query_res[i] = json.loads(val)
         return query_res
+
+    async def return_session_info(self, sess_uuid, snare_uuid= None):
+        query_res = []
+        if snare_uuid:
+            snare_uuids = [snare_uuid]
+        else:
+            snare_uuids = await self.return_snares()
+
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            if sessions == 'Invalid SNARE UUID':
+                continue
+            for sess in sessions:
+                if sess['sess_uuid'] == sess_uuid:
+                    return sess
+
+    async def return_sessions(self, filters):
+        query_res = []
+        if 'snare_uuid' in filters:
+            snare_uuids = [filters['snare_uuid']]
+        else:
+            snare_uuids = await self.return_snares()
+
+        matching_sessions = []
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            for sess in sessions:
+                if sessions == 'Invalid SNARE UUID':
+                    continue
+                match_count = 0
+                if 'user-agent' in filters:
```

Comment:
Just as a quick example, how it could be done with operators
 
```
    def applay_filter(self, sessions, filter_name, value, op):
        ops = {"in":operator.contains,
               "ge":operator.ge,
               "lt":operator.le,
               "eq":operator.eq}
        result = []
        for sess in sessions:
            if sessions == 'Invalid SNARE UUID':
                continue
            try:
                if ops[op](sess[filter_name],value):
                    result.append(sess)
            except KeyError:
                raise

        return result

    async def return_sessions(self, filters):
        avaliable_filters = {"user_agent":"in",
                             "peer_ip":"eq",
                             "attack_type":"in",
                             "start_time":"le",
                             "end_time":"ge",
                             "possible_owners":"in"}

        if 'snare_uuid' in filters:
            snare_uuids = [filters['snare_uuid']]
        else:
            snare_uuids = await self.return_snares()

        for snare_id in snare_uuids:
            sessions = await self.return_snare_info(snare_id)
            result = sessions
            for f_name, value in filters.items():
                try:
                    result = self.applay_filter(result, f_name,value,avaliable_filters[f_name])
                except KeyError:
                    return "invalid filter or parameter"
                else:
                    return [s['sess_uuid'] for s in result]
```
I think we can make a **separate task** for filters design and find the best way to do this task.

But still, I strongly recommend you to use `try-except` in your code :)